### PR TITLE
Populate import counts during applies and clean up output

### PR DESCRIPTION
**Pull request by radditude**
There's no actual action that happens when applying an import, so in order to populate a count of imported objects and include some helpful output about what's happening, we'll use the slightly awkward method of iterating over the planned resource changes during an apply and manually calling a hook (similar to the approach discussed in #33155). 

This PR also establishes new `PrePlanImport`, `PostPlanImport`, `PreApplyImport`, and `PostApplyImport` hooks for the new configuration-driven import mechanism, so we'll no longer need to reuse the import-related hooks called during the legacy import command.

Output (import only):

```
tfe_organization.name: Preparing import... [id=boring-test-org]
tfe_organization.name: Refreshing state... [id=boring-test-org]

Terraform will perform the following actions:

  # tfe_organization.name will be imported
    resource "tfe_organization" "name" {
        allow_force_delete_workspaces                           = false
        assessments_enforced                                    = false
        collaborator_auth_policy                                = "password"
        cost_estimation_enabled                                 = true
        default_project_id                                      = "prj-12345"
        email                                                   = "email@example.com"
        id                                                      = "boring-test-org"
        name                                                    = "boring-test-org"
        send_passing_statuses_for_untriggered_speculative_plans = true
        session_remember_minutes                                = 0
        session_timeout_minutes                                 = 0
    }

Plan: 1 to import, 0 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

tfe_organization.name: Importing... [id=boring-test-org]
tfe_organization.name: Import complete [id=boring-test-org]

Apply complete! Resources: 1 imported, 0 added, 0 changed, 0 destroyed.
```

Output (import combined with update):

```
tfe_organization.name: Preparing import... [id=boring-test-org]
tfe_organization.name: Refreshing state... [id=boring-test-org]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following
symbols:
  ~ update in-place

Terraform will perform the following actions:

  # tfe_organization.name will be updated in-place
  # (imported from "boring-test-org")
  ~ resource "tfe_organization" "name" {
      ~ allow_force_delete_workspaces                           = false -> true
        assessments_enforced                                    = false
        collaborator_auth_policy                                = "password"
        cost_estimation_enabled                                 = true
        default_project_id                                      = "prj-12345"
        email                                                   = "email@example.com"
        id                                                      = "boring-test-org"
        name                                                    = "boring-test-org"
        send_passing_statuses_for_untriggered_speculative_plans = true
        session_remember_minutes                                = 0
        session_timeout_minutes                                 = 0
    }

Plan: 1 to import, 0 to add, 1 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

tfe_organization.name: Importing... [id=boring-test-org]
tfe_organization.name: Import complete [id=boring-test-org]
tfe_organization.name: Modifying... [id=boring-test-org]
tfe_organization.name: Modifications complete after 0s [id=boring-test-org]

Apply complete! Resources: 1 imported, 0 added, 1 changed, 0 destroyed.
```

## Target Release

1.5.x

### CODE REVIEW

**Comment by kmoe** [comment]
I think the old and new models are sufficiently different to warrant new hooks.

Let's err on the side of verbosity in making sure the legacy code path is clearly delimited.

In particular, as of https://github.com/hashicorp/terraform/pull/33155/files#diff-336af58f42294b507a0730c062969ef7ab365459f6976ae5fa445e65acb8ac55R297 we have two versions of the `PreImportState` hook, one for OperarationImport and one for OperationPlan. It may not be obvious to future devs that these represent the import CLI and config-driven import workflows respectively.

It would be more explicit to have the full set of hooks be as follows:

### Legacy import behaviour (unchanged)
Hooks run during `Import` operation:
 - `PreImportState`
 - `PostImportState`

### New plannable import behaviour
Hooks run during `Plan` operation:
 - `PrePlanImport`
 - `PostPlanImport`

Hooks run during `Apply` operation:
 - `PreApplyImport`
 - `PostApplyImport`

If we do this, we also no longer need to pass the `operation` to `m.uiHook` (added in https://github.com/hashicorp/terraform/pull/33155/files#diff-6627e3489968d07107612d76f924dc2cfc0aa526715a5fef572f55560f6e1912R608).

@liamcervante What do you think?

**Comment by liamcervante** [comment]
Yeah, sounds good to me.

I can revert https://github.com/hashicorp/terraform/pull/33155, and we could maybe just do it all in this PR?

**Comment by kmoe** [comment]
Yep, approving #33171 and will let @radditude decide whether to merge that or just continue here.

File: internal/terraform/context_apply.go [hunk 1]
```diff
@@ -38,6 +38,17 @@ func (c *Context) Apply(plan *plans.Plan, config *configs.Config) (*states.State
 		return nil, diags
 	}
 
+	for _, rc := range plan.Changes.Resources {
+		// import is a no-op change, but we'd like to show some helpful output that mirrors
+		// the way we show other changes.
+		if rc.Importing != nil {
+			for _, h := range c.hooks {
+				h.PreApplyImport(rc.Addr, *rc.Importing)
+				h.PostApplyImport(rc.Addr, *rc.Importing)
```
Comment (radditude) [inline internal/terraform/context_apply.go:47]: It feels a bit silly to call both hooks here, but I couldn't find a better place to call `PostApplyImport` 😅 should we keep the two separate pre/post hooks anyway for forward-thinking purposes?

File: internal/command/views/hook_ui.go [hunk 2]
```diff
@@ -304,6 +304,33 @@ func (h *UiHook) PostImportState(addr addrs.AbsResourceInstance, imported []prov
 	return terraform.HookActionContinue, nil
 }
 
+func (h *UiHook) PrePlanImport(addr addrs.AbsResourceInstance, importID string) (terraform.HookAction, error) {
+	h.println(fmt.Sprintf(
+		h.view.colorize.Color("[reset][bold]%s: Preparing import... [id=%s]"),
```
Comment (radditude) [inline internal/command/views/hook_ui.go:309]: Currently there's no output for the `PostPlanImport` hook, but I'm open to adding some!

File: internal/terraform/context_apply.go (hunk 1 above)
Comment (liamcervante) [reply internal/terraform/context_apply.go:47]: I'm okay with keeping two to ensure that it's future proofed, maybe just add a comment explaining why we we're calling both immediately?

Comment (radditude) [reply internal/terraform/context_apply.go:49]: Makes sense to me!

**Review by liamcervante — APPROVED** [review]

**Pull request merged**